gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java

```diff
@@ -3455,18 +3455,17 @@ public void checkDeletionOrderForHnBucketSkewedFolders() throws InterruptedExcep
   /** Helper function to create a balanced set of folder resources */
   private void addFolders(List<FolderInfo> foldersToDelete, String curFolderName) {
     Random r = new Random();
-    String abc = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
     Queue<String> q = new ArrayDeque<>();
     q.add(curFolderName);
 
     while (!q.isEmpty()) {
       String top = q.poll();
       foldersToDelete.add(new FolderInfo(FolderInfo.createFolderInfoObject(BUCKET_NAME, top)));
-      if (foldersToDelete.size() > 30) return;
+      if (foldersToDelete.size() > 2000) return;
```

Comment:
why is this flaky and how this fixes it?

Reply:
This line is Not flaky. Line https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/1171/files#diff-083531fc5918e3d31d283634e518c0f9cc52750c888ed8a7c2d97e0109c8898cR3467 is flaky.

Reply:
Essentially, `addFolders` function could generate multiple folders with same name, ie two folders with name `bucket/A/B`. This is not supported by actual HNS bucket and thus was breaking the code.